### PR TITLE
fileupload-cancel-btn

### PR DESCRIPTION
**Pull request by roggc**
add cancel button to fileupload component

https://drive.google.com/file/d/1vmNjkmsGTXz7vEB0lmtjdLo6Qfu_nHe4/view?usp=sharing

### CODE REVIEW

File: src/Containers/FileUpload/FileUpload.tsx [hunk 1]
```diff
@@ -488,39 +489,28 @@ export const FileUpload: React.FC<IFileUploadProps> = ({
...
-                            if(isMounted.current){
-                                setIsSuccess(false)
-                            }
-                        },messageDuration)
-                    }
+            const worker=new Worker('./worker.js', { type: 'module' })
+            workerRef.current=worker
+            worker.postMessage({file})
+            worker.onmessage=(e)=>{
+                const {base64StringFile}=e.data
```
Comment (ralph-dev) [inline src/Containers/FileUpload/FileUpload.tsx:496]: {data}
Then
Const {base64} = data

Works as well but your way is fine

File: src/Containers/FileUpload/FileUpload.tsx [hunk 2]
```diff
@@ -488,39 +489,28 @@ export const FileUpload: React.FC<IFileUploadProps> = ({
...
+                if(base64StringFile!==undefined){
```
Comment (ralph-dev) [inline src/Containers/FileUpload/FileUpload.tsx:497]: What if it equals null?
Use 
!Var

**Review by ralph-dev — APPROVED** [review]
Lgtm, nice job this is a very important concept to pick up when doing intensive tasks on the web!

File: src/Containers/FileUpload/FileUpload.tsx (hunk 2 above)
Comment (roggc) [reply src/Containers/FileUpload/FileUpload.tsx:497]: but what if is an empty string because file read was empty?

Comment (roggc) [reply src/Containers/FileUpload/FileUpload.tsx:497]: maybe and can use ??

Comment (roggc) [reply src/Containers/FileUpload/FileUpload.tsx:497]: I don't think it can be null... can be undefined or empty string (or normal string)

**Pull request merged**